### PR TITLE
Make all tables in fastlane more responsive to the terminal width

### PR DESCRIPTION
**Pull request by KrauseFx**
Moved from https://github.com/fastlane/fastlane/pull/7391 by mistake

The design on the top table is the one that's used
screen shot 2017-04-06 at 7 06 57 pm

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/print_table.rb [hunk 1]
```diff
@@ -34,16 +40,68 @@ def print_values(config: nil, title: nil, hide_keys: [], mask_keys: [])
...
+        # -4 per column - as tt adds "| " and " |"
+        terminal_table_padding = 4
+        max_length = number_of_cols - (col_count * terminal_table_padding)
+
+        max_value_length = (max_length / col_count)
+
+        return_array = rows.map do |row|
+          row.map do |col|
+            if col.nil?
+              col # we want to keep the nil and not convert it to a string
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/print_table.rb]: Anybody knows how can I early break out of the `map` with a specific value? This way we'd fix the last remaining code style issue before this is ready 👍

**Comment by KrauseFx** [comment]
Quick question @hjanuschka: how does this behave on CI systems? Should we just skip all transformations when `Helper.ci?` is true?

**Comment by hjanuschka** [comment]
right now it only skips when `FL_SKIP_TABLE_TRANSFORM` env is set.
i am not sure - i remember that @DanToml  mentioned that terminal width on circle is low - so yeah i guess - skipping on `is_ci?` might be  a good idea.

File: fastlane_core/lib/fastlane_core/print_table.rb (hunk 1 above)
Comment (DanToml) [inline fastlane_core/lib/fastlane_core/print_table.rb]: The best way would be to break this out into a transform_row function

**Review by endocrimes — COMMENTED** [review]
I'd force disable this on CI but outside of that it looks good.

File: fastlane_core/lib/fastlane_core/print_table.rb (hunk 1 above)
Comment (mfurtak) [inline fastlane_core/lib/fastlane_core/print_table.rb]: Agreed about extracting a method here. You could also consider starting with

```ruby
return col if col.nil?
```

to allow you to remove the outer if/else.

**Comment by KrauseFx** [comment]
Thanks everyone, I updated the PR to address the feedback 👍

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
FastlaneCore FastlaneCore::ConfigurationFile Properly loads and handles various configuration files Prints out a table of summary prints out a table of all the set values | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/configuration_file_spec.rb'>configuration_file_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable prints out all the information in a nice table | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable automatically masks sensitive options | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable supports printing default values and ignores missing unset ones  | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable Breaks down lines middle truncate | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable Breaks down lines newline | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable Breaks down lines raises an exception for invalid transform | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
Pilot::TesterManager prints tester lists when invoked from a global context prints a table with columns including device and version info | <a href='https://github.com/fastlane/fastlane/blob/f3d3d5e6886bcf6ff5a4cc68a7418fd4dc221f16/./pilot/spec/tester_manager_spec.rb'>tester_manager_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
FastlaneCore FastlaneCore::ConfigurationFile Properly loads and handles various configuration files Prints out a table of summary prints out a table of all the set values | <a href='https://github.com/fastlane/fastlane/blob/7bb28b4f19ac48124f8b245b411030f74ff22eac/./fastlane_core/spec/configuration_file_spec.rb'>configuration_file_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable prints out all the information in a nice table | <a href='https://github.com/fastlane/fastlane/blob/7bb28b4f19ac48124f8b245b411030f74ff22eac/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable automatically masks sensitive options | <a href='https://github.com/fastlane/fastlane/blob/7bb28b4f19ac48124f8b245b411030f74ff22eac/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
FastlaneCore FastlaneCore::PrintTable supports printing default values and ignores missing unset ones  | <a href='https://github.com/fastlane/fastlane/blob/7bb28b4f19ac48124f8b245b411030f74ff22eac/./fastlane_core/spec/print_table_spec.rb'>print_table_spec.rb</a>|
Pilot::TesterManager prints tester lists when invoked from a global context prints a table with columns including device and version info | <a href='https://github.com/fastlane/fastlane/blob/7bb28b4f19ac48124f8b245b411030f74ff22eac/./pilot/spec/tester_manager_spec.rb'>tester_manager_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Pilot::TesterManager prints tester lists when invoked from a global context prints a table with columns including device and version info | <a href='https://github.com/fastlane/fastlane/blob/b0646cb58d216ace4eadaf78adcaec0ecddfe54f/./pilot/spec/tester_manager_spec.rb'>tester_manager_spec.rb</a>|
Pilot::TesterManager prints tester lists when invoked from the context of an app prints a table without columns showing device and version info | <a href='https://github.com/fastlane/fastlane/blob/b0646cb58d216ace4eadaf78adcaec0ecddfe54f/./pilot/spec/tester_manager_spec.rb'>tester_manager_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Pilot::TesterManager prints tester lists when invoked from a global context prints a table with columns including device and version info | <a href='https://github.com/fastlane/fastlane/blob/b0646cb58d216ace4eadaf78adcaec0ecddfe54f/./pilot/spec/tester_manager_spec.rb'>tester_manager_spec.rb</a>|
Pilot::TesterManager prints tester lists when invoked from the context of an app prints a table without columns showing device and version info | <a href='https://github.com/fastlane/fastlane/blob/b0646cb58d216ace4eadaf78adcaec0ecddfe54f/./pilot/spec/tester_manager_spec.rb'>tester_manager_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request closed without merging**